### PR TITLE
Improve EmptyBuildItem handling and add a note on how to use them

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 1]
```diff
@@ -755,6 +755,9 @@ Empty build items are final (usually empty) classes which extend `io.quarkus.bui
 They represent build items that don't actually carry any data, and allow such items to be produced and consumed
 without having to instantiate empty classes.  They cannot themselves be instantiated.
 
+IMPORTANT: As they cannot be instantiated, they cannot be injected by any means, nor be returned by a build step (or via a `BuildProducer`).
+To produce an empty built item you must annotate the build step with `@Produce(MyEmptyBuildItem.class)` and to consume it by `@Consume(MyEmptyBuildItem.class)`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-extensions.adoc:759]:
```suggestion
To produce an empty build item you must annotate the build step with `@Produce(MyEmptyBuildItem.class)` and to consume it by `@Consume(MyEmptyBuildItem.class)`.
```

**Review by gsmet — DISMISSED** [review]
Spotted a typo. I will let others check the rest :).

**Comment by loicmathieu** [comment]
Thanks @gsmet, Georgios works with me on this one ;)

**Review by geoand — APPROVED** [review]
Great!

**Pull request merged**